### PR TITLE
Popover: render in scroll container

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

This PR fixes popover z-index issues by rendering popovers in a slot within the closest scrollable parent.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

File: packages/edit-post/src/components/editor-regions/index.js [hunk 1]
```diff
@@ -31,7 +31,9 @@ function EditorRegions( { footer, header, sidebar, content, publish, className }
 					aria-label={ __( 'Editor content' ) }
 					tabIndex="-1"
 				>
-					{ content }
+					<Popover.Slot>
```
Comment (youknowriad) [inline packages/edit-post/src/components/editor-regions/index.js]: What's the reasoning for including this in `EditorRegions`? Should we remove the existing PopoverSlot included in the Layout?

File: packages/components/src/popover/index.js [hunk 2]
```diff
@@ -483,6 +485,24 @@ const Popover = ( {
 
 const PopoverContainer = Popover;
 
-PopoverContainer.Slot = () => <Slot bubblesVirtually name={ SLOT_NAME } />;
```
Comment (youknowriad) [inline packages/components/src/popover/index.js]: I'm not sure I understand why the change here and why we need multiple Popover Slots?

**Comment by jasmussen** [comment]
It's hard to review this as it doesn't include the block toolbars in a popover, so there's no way to easily reproduce the issue it's meant to fix. However, to expand on your description, in your other PR the block toolbar becomes a popover, so its node exists in a different stacking context than that of the top toolbar. When that happens, the block toolbar can in some situations overlap the block toolbar, when it should be under it, like this:

Screenshot 2019-12-10 at 09 43 22

By moving the block toolbar into the same stacking context as the top toolbar, you can use `z-index` to control the depth. You can't otherwise.

File: packages/components/src/popover/index.js (hunk 2 above)
Comment (ellatrix) [reply packages/components/src/popover/index.js]: `Popovers` should ideally be rendered in the nearest scrollable parent container so that the z-indices don't break. So Popovers with anchors in the content will render within the content scroll container, and Popovers in e.g. the header will render in the document scroll area. @jasmussen does this make sense to you?

**Comment by ellatrix** [comment]
Right, it's probably better to wait with this until #19024 is merged, I just wanted to make sure I have a fix lined up for the z-index issue.

**Comment by ellatrix** [comment]
This is no longer needed since #18779 now puts the toolbar in the right place.
